src/checker/http_checker.rs

```diff
@@ -219,25 +225,29 @@ impl Checker for HttpChecker {
                         status_type: CheckStatusReasonType::DnsError,
                         description: message,
                     }
-                } else if let Some(message) = ssl_error(&e) {
+                } else if let Some(message) = tls_error(&e) {
                     CheckStatusReason {
-                        status_type: CheckStatusReasonType::Failure,
+                        status_type: CheckStatusReasonType::TlsError,
                         description: message,
                     }
                 } else if let Some(message) = connection_error(&e) {
                     CheckStatusReason {
-                        status_type: CheckStatusReasonType::Failure,
+                        status_type: CheckStatusReasonType::ConnectionError,
                         description: message,
                     }
-                } else if let Some(message) = hyper_error(&e) {
+                } else if let Some((status_type, message)) = hyper_error(&e) {
                     CheckStatusReason {
-                        status_type: CheckStatusReasonType::Failure,
+                        status_type,
                         description: message,
                     }
                 } else {
+                    // if any error falls through we should log it,
+                    // none should be.
```

Comment:
```suggestion
                    // if any error falls through we should log it,
                    // none should fall through.
```

This is what this meant right?